File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/data_structures/test_stack.java
// test_stack.java

public class test_stack
{
	public static <T extends Object> void display_stack(dp_stack<T> a_stack)
	{
		System.out.print("display_stack(a_stack): <top> ");
		while (!a_stack.is_empty())
		{
			T a_value = a_stack.top();
			System.out.print(a_value + " ");
			a_stack.pop();
		}
		System.out.println("<bottom>");
	}

	public static void main(String[] args)
	{
		dp_stack<Integer> a_stack = new dp_stack<Integer>();
		display_stack(a_stack);

		a_stack.push(1);
		display_stack(a_stack);

		a_stack.push(1);
		a_stack.push(2);
		display_stack(a_stack);
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/data_structures/testStack.m
% testStack.m

function testStack
	aStack = DpStack;
	displayStack(aStack);

	aStack.push(1);
	displayStack(aStack);

	aStack.push(1);
	aStack.push(2);
	displayStack(aStack);
end

function displayStack(aStack)
	fprintf('displayStack(aStack): <top> ');
	while ~aStack.isEmpty
		aValue = aStack.top;
		fprintf('%d ', aValue);
		aStack.pop;
	end
	fprintf('<bottom>\n');
end
